### PR TITLE
Added new exports manifests and a define off of this to generate the …

### PR DESCRIPTION
**Pull request by vamegh**
…exports file for nfs server exports, along with appropriate templates for this.

	new file:   manifests/exports.pp
	new file:   manifests/exports_data.pp
	modified:   manifests/server.pp
	new file:   templates/exports_body.epp
	new file:   templates/exports_header.epp

### CODE REVIEW

**Comment by ghoneycutt** [comment]
Hi @vamegh 

Thanks for sending code. Everyone I know using this has dedicated devices like NetApp's to serve the NFS exports so I really appreciate you looking at managing the exports in a more dynamic format. Comments are inline with the code.

File: manifests/exports.pp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      group   => "$group",
+      mode    => "$mode",
+      order   => 'numeric',
+    }
+    # template vars:
+      # * message
+    concat::fragment { "${config_name}-general}":
+      target  => "$config_name",
+      order   => 0,
+      content => epp("${module_name}/exports_header.epp",{message => $message}),
```
Comment (ghoneycutt) [inline manifests/exports.pp]: Please use erb as this code must remain compatible with Puppet v3

File: manifests/exports.pp [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class nfs::exports (
```
Comment (ghoneycutt) [inline manifests/exports.pp]: Shouldn't this just be part of the server class?

File: manifests/exports_data.pp [hunk 3]
```diff
@@ -0,0 +1,20 @@
+define nfs::exports_data (
+  $config_title = "$title",
```
Comment (ghoneycutt) [inline manifests/exports_data.pp]: I think you mean to use $name here. We don't need another variable just to change the name of the variables around.

File: manifests/exports_data.pp [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  $message = 'This File is managed by Puppet - NFS Module',
```
Comment (ghoneycutt) [inline manifests/exports_data.pp]: The header should be static and non-configurable. We use the first two lines from https://github.com/ghoneycutt/puppet-module-nfs/blob/master/files/exports#L1

File: manifests/exports_data.pp [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  $path = '',
```
Comment (ghoneycutt) [inline manifests/exports_data.pp]: shouldn't this be a mandatory parameter without a default?

File: manifests/exports_data.pp [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  $host = '',
```
Comment (ghoneycutt) [inline manifests/exports_data.pp]: shouldn't this be a mandatory parameter without a default?

File: templates/exports_body.epp [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<% if !empty($options) { -%>
```
Comment (ghoneycutt) [inline templates/exports_body.epp]: Something like this

``` erb
<%= @path %>  <%= @host %><% if @options != nil -%>(<%= @options.join(',') -%>)<% end -%>
```

**Comment by ghoneycutt** [comment]
We also need the new define to be documented including all of its parameters and to have comprehensive spec tests.

**Comment by vamegh** [comment]
Hi Garett,

This is just something for you to use if you want, or if some one else
wants for puppet 4. I have just added it to make life easier -- exports.pp
can be merged with server.pp, I kept them separate initially to not mess
with your original manifests, which I ended up modifying anyway to call it.

$name and $title are essentially the same, however in puppet docs, it
states the following:

```$title and $name

Every defined type gets two “free” parameters, which are always available
and do not have to be explicitly added to the definition:

$title is always set to the title of the instance. Since it is guaranteed
to be unique for each instance, it is useful when making sure that
contained resources are unique. (See “Resource Uniqueness” below.)
$name defaults to the value of $title, but users can optionally specify a
different value when they declare an instance. This is only useful for
mimicking the behavior of a resource with a namevar, which is usually
unnecessary. If you are wondering whether to use $name or $title, use
$title.
https://docs.puppetlabs.com/puppet/latest/reference/lang_defined_types.html
 ```

just as an FYI.

Again the banner message, it just provides choice to manage the banner in
hiera.

You can make the options into an array, but you can just pass along all of
the parameters as a string (I did this really quickly)

I hacked this together in about an hour to test something we were working
on in aws, we are not even using this, but I thought it might save you some
time, since you had it commented as a to-do. I've tested it on puppet4 and
it generates the file correctly.

If I get time I'll come back and convert it to an erb and change some of
the stuff you mentioned above, but if you want just grab it and do whatever
you want with it, if it'll save you some time.

Kind Regards,

-v

**Comment by vamegh** [comment]
Hey, 

most of the stuff you wanted changed done. 

-v

**Pull request closed without merging**